src/Microsoft.DocAsCode.MarkdigEngine.Extensions/MarkdownContext.cs

```diff
@@ -41,14 +46,36 @@ public class MarkdownContext
         /// </summary>
         public GetLinkDelegate GetLink { get; }
 
+        /// <summary>
+        /// Log warning
+        /// </summary>
+        public LogActionDelegate LogWarning { get; }
+
+        /// <summary>
+        /// Log error
+        /// </summary>
+        public LogActionDelegate LogError { get; }
+
+        /// <summary>
+        /// Set logger scope function
+        /// </summary>
+        public Func<string, IDisposable> SetLoggerScope { get; }
+
         public MarkdownContext(
             IReadOnlyDictionary<string, string> tokens,
+            LogActionDelegate logWarning,
+            LogActionDelegate logError,
+            Func<string, IDisposable> setLoggerScope,
```

Comment:
can we remove setLoggerScope?

Reply:
why? scope will make log easier to locate

---
In reply to: [187253456](https://github.com/dotnet/docfx/pull/2753#discussion_r187253456) [](ancestors = 187253456)